src/app/services/resource.service.ts

```diff
@@ -287,33 +290,44 @@ export class ResourceService {
 
   // -------------------------------------- data table ---------------------------------------------
   getDataTable(dataTable: DataTable): Observable<DataTable> {
-    let isUsingHeaders = dataTable.isUsingHeaders;
     let offset = dataTable.offset;
     let limit = dataTable.limit;
 
+    // Fetch dimensions for the data matching the constraint
     return this.getDimensions(dataTable.constraint)
       .switchMap((transmartStudyDimensions: TransmartStudyDimensions) => {
+        // Merge available dimensions and current table state
         let tableState: TransmartTableState =
-          TransmartMapper.mapStudyDimensionsToTableState(transmartStudyDimensions, dataTable);
+          TransmartDataTableMapper.mapStudyDimensionsToTableState(transmartStudyDimensions, dataTable);
```

Comment:
still doubting the motivation behind separating mappers into even more mappers, if we worry about the transmart mapper getting large, we don't have to be. the mapper functions are just stand-alone, static functions, distinguishing them with comment lines would be enough. Ideally would prefer to keep all the mapper function in one place.

Reply:
That is not a proper way to organise your code.

Reply:
or find alternative ways to hide these other mappers from resource service